dot/sync/test_helpers.go

```diff
@@ -51,3 +70,201 @@ func (bp *mockBlockProducer) Resume() error {
+	require.NoError(t, err)
+	return gen, genTrie, genesisHeader
+}
+
+// BuildBlock ...
+func BuildBlock(t *testing.T, srv *Service, parent *types.Header, ext types.Extrinsic) *types.Block {
+	header := &types.Header{
+		ParentHash: parent.Hash(),
+		Number:     big.NewInt(0).Add(parent.Number, big.NewInt(1)),
+		Digest:     types.Digest{},
+	}
+
+	instance := srv.runtime
+	err := instance.InitializeBlock(header)
+	require.NoError(t, err)
+
+	idata := types.NewInherentsData()
+	err = idata.SetInt64Inherent(types.Timstap0, uint64(time.Now().Unix()))
+	require.NoError(t, err)
+
+	err = idata.SetInt64Inherent(types.Babeslot, 1)
+	require.NoError(t, err)
+
+	err = idata.SetBigIntInherent(types.Finalnum, big.NewInt(0))
+	require.NoError(t, err)
+
+	ienc, err := idata.Encode()
+	require.NoError(t, err)
+
+	// Call BlockBuilder_inherent_extrinsics which returns the inherents as extrinsics
+	inherentExts, err := instance.InherentExtrinsics(ienc)
+	require.NoError(t, err)
+
+	// decode inherent extrinsics
+	exts, err := scale.Decode(inherentExts, [][]byte{})
+	require.NoError(t, err)
+
+	inExt := exts.([][]byte)
+
+	var body *types.Body
+	if ext != nil {
+		var txn *transaction.Validity
+		txn, err = instance.ValidateTransaction(append([]byte{byte(types.TxnExternal)}, ext...))
+		require.NoError(t, err)
+
+		vtx := transaction.NewValidTransaction(ext, txn)
+		_, err = instance.ApplyExtrinsic(ext) // TODO: Determine error for ret
+		require.NoError(t, err)
+
+		body, err = babe.ExtrinsicsToBody(inExt, []*transaction.ValidTransaction{vtx})
+		require.NoError(t, err)
+
+	} else {
+		body = types.NewBody(inherentExts)
+	}
+
+	// apply each inherent extrinsic
+	for _, ext := range inExt {
+		in, err := scale.Encode(ext) //nolint
+		require.NoError(t, err)
+
+		ret, err := instance.ApplyExtrinsic(in)
+		require.NoError(t, err)
+		require.Equal(t, ret, []byte{0, 0})
+	}
+
+	res, err := instance.FinalizeBlock()
+	require.NoError(t, err)
+	res.Number = header.Number
+
+	return &types.Block{
+		Header: res,
+		Body:   body,
+	}
+}
+
+// CreateExtrinsic ...
+func CreateExtrinsic(t *testing.T, srv *Service) []byte {
+	t.Helper()
+	rawMeta, err := srv.runtime.Metadata()
+	require.NoError(t, err)
+	decoded, err := scale.Decode(rawMeta, []byte{})
+	require.NoError(t, err)
+
+	metaData := &subtypes.Metadata{}
+	err = subtypes.DecodeFromBytes(decoded.([]byte), metaData)
+	require.NoError(t, err)
+
+	rv, err := srv.runtime.Version()
+	require.NoError(t, err)
+
+	bob, err := subtypes.NewAddressFromHexAccountID("0x90b5ab205c6974c9ea841be688864633dc9ca8a357843eeacf2314649965fe22")
+	require.NoError(t, err)
+
+	call, err := subtypes.NewCall(metaData, "Balances.transfer", bob, subtypes.NewUCompactFromUInt(123450000000000))
+	require.NoError(t, err)
+
+	// Create the extrinsic
+	ext := subtypes.NewExtrinsic(call)
+	genesisHash, err := subtypes.NewHashFromHexString("0x64597c55a052d484d9ff357266be326f62573bb4fbdbb3cd49f219396fcebf78")
```

Comment:
just curious about where this value comes from?

Reply:
This is genesis hash. Updated the code to take it from blockstore.